tools/tokenserver/test_database.py

```diff
@@ -162,20 +162,32 @@ def test_cleanup_of_old_records(self):
         # Do a sleep halfway through so we can test use of grace period.
         email1 = 'test1@mozilla.com'
         user1 = self.database.allocate_user(email1)
+        # We have to sleep between every user create/update operation: if two
```

Comment:
I do wonder if this might eventually bite us (e.g. if the client is also in rust and is on a particularly close node, they may respond faster than one ms.) We might want to add an issue to check for active user records before writing a second and returning a 409 CONFLICT if one is detected.

Reply:
I'm not sure that the risk of this happening on the new Tokenserver is any greater than the risk of it happening on the old Tokenserver 🤔 Requests on the new Tokenserver will still take well over 1 ms to complete, so I _think_ the only way this could happen would be if a client decided to bombard us with requests, and I think the old Tokenserver is just as vulnerable to that